src/clj/rems/application/search.clj

```diff
@@ -29,9 +32,48 @@
 
 (defn- index-application! [^IndexWriter writer app]
   (log/info "Indexing application" (:application/id app))
-  (let [doc (Document.)]
-    (.add doc (StringField. "id" (str (:application/id app)) Field$Store/YES))
-    (.add doc (TextField. "applicant" (str (:application/applicant app)) Field$Store/NO))
+  (let [doc (Document.)
+        id (->> [(:application/id app)
+                 (:application/external-id app)]
+                (str/join " "))
+        applicant (->> [(:application/applicant app)
+                        (application-util/get-applicant-name app)
+                        (:mail (:application/applicant-attributes app))]
+                       (str/join " "))
+        member (->> (:application/members app)
+                    (mapcat (fn [member]
+                              [(:userid member)
+                               (application-util/get-member-name member)
+                               (:mail member)]))
+                    (str/join " "))
+        title (:application/description app)
+        resource (->> (:application/resources app)
+                      (mapcat (fn [resource]
+                                (vals (:catalogue-item/title resource))))
+                      (str/join " "))
+        state (->> (:languages env)
```

Comment:
nit: how about a `for-all-languages` helper?

Reply:
Let's wait until the code is duplicated somewhere. This far there is only one.